### PR TITLE
Fix units on cache expiry

### PR DESCRIPTION
**Pull request by michaelabuckley**
Added a test to abuse Caffeine

### CODE REVIEW

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/5_5_0/2758-fix-cache-expiry-units.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+type: fix
+issue: 2758
+title: "The internal cache for tags, concepts, and others had longer or shorter expiry times that was intended.
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/5_5_0/2758-fix-cache-expiry-units.yaml:4]:
```suggestion
title: "The internal cache for tags, concepts, and others had longer or shorter expiry times than was intended.
```

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/util/MemoryCacheService.java [hunk 2]
```diff
@@ -69,16 +67,16 @@ public void start() {
 			switch (next) {
 				case CONCEPT_TRANSLATION:
 				case CONCEPT_TRANSLATION_REVERSE:
-					timeoutSeconds = myDaoConfig.getTranslationCachesExpireAfterWriteInMinutes() * 1000;
+					timeoutSeconds = SECONDS.convert(myDaoConfig.getTranslationCachesExpireAfterWriteInMinutes(), MINUTES);
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/util/MemoryCacheService.java:70]: hahahah how did this happen! 1000 * however many minutes == seconds. I assume the original implementation thought it was milis and not minutes.

**Review by tadgh — APPROVED** [review]
Looks good, approved. Does this actually solve our tag cache issue though? The tests seem to exercise it, but I don't see how the cache timeout changes actually affect the outcome. Was this set of tests to just prove the functionality of the cache under stress?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/util/MemoryCacheService.java (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/util/MemoryCacheService.java:70]: I have no idea, but this was the line that made be bother to make an MR,

**Comment by michaelabuckley** [comment]
> Looks good, approved. Does this actually solve our tag cache issue though? The tests seem to exercise it, but I don't see how the cache timeout changes actually affect the outcome. Was this set of tests to just prove the functionality of the cache under stress?

It doesn't.  The @Disable'd test shows that computeIfAbsent() confirms that we will lock the ConcurrentHashMap while growing the cache, which was the working theory.

**Pull request merged**